rpi/sensors/temp_management.py

```diff
@@ -0,0 +1,216 @@
+import RPi.GPIO as GPIO
+import time
+
+import RPi.GPIO as GPIO
+import matplotlib.pyplot as plt
+from matplotlib.animation import FuncAnimation
+
+from rpi.sensors.thermometer import ThermometerList
+
+
+# import numpy as np
+
+
+class GetCurrentTemp():
+    def __init__(self):
+        # ID's of sensors to poll - change when lay out finalized.
+        sensor_id_list = ['1', '2', '3', '4']
+
+        # call thermometer thread to retrieve current temperature
+        current_temp_val = ThermometerList.get_temperature_data(sensor_id_list)
+        print(f'Current temperature retrieved from thermometer thread: {current_temp_val}')
+
+        GetCurrentTemp.current_avg_temp(self, current_temp_val)
+
+    def current_avg_temp(self, current_temps):
+        sum = 0
+        count = 0
+        for key, value in current_temps.items():
+            # handle 'none' passed in for cases where sensor is disconnected
+            try:
+                print(value)
+                sum += value
+                count += 1
+            except TypeError:
+                print(f"Not all sensors are providing acceptable temperature data. "
+                      f"Cannot perform designated operation on sensor: ({key} | value: '{value}') - Discarded")
+                pass
+
+        avg_temp = sum / count
+
+        print(f'Current average temperature : {avg_temp}')
+        return avg_temp
+
+
+class PID():
+    def __init__(self, P, I, D, current_time=None):
+        #pid initialization
+        self.Kp = P
+        self.Ki = I
+        self.Kd = D
+        self.sample_time = 0.00
+        self.current_time = current_time if current_time is not None else time.time()
+        self.last_time = self.current_time
+        self.clear()
+
+    def clear(self):
+        """ 
+        Clear PID computations and coefficients
+        """
+        self.SetPoint = 35 #desired temp
```

Comment:
This could maybe be set as a constant at the top of this file? This way, it's easy to see how to change the desired temperature.

Reply:
SetPoint made into a global variable at top of file.